processing/src/main/java/io/druid/query/metadata/SegmentMetadataQueryQueryToolChest.java

```diff
@@ -149,7 +149,7 @@ public SegmentAnalysis apply(SegmentAnalysis arg1, SegmentAnalysis arg2)
 
             List<Interval> newIntervals = null;
             if (query.hasInterval()) {
-              newIntervals = arg1.getIntervals();
+              newIntervals = new ArrayList<>(arg1.getIntervals());
```

Comment:
it is just that the List returned by arg1.getIntervals() is immutable , so it was failing with UnsupportedOperationException. I was probably biased and "thought" that the explanation was trivial so dint add the comment... will add it to make it clear.